src/proof_system/linearisation_poly.rs

```diff
@@ -74,6 +74,7 @@ pub(crate) struct ProofEvaluations<F: PrimeField> {
 }
 
 /// Compute the linearisation polynomial.
+#[allow(clippy::type_complexity)] // FIXME: Make a separate struct for the tuple arguments.
```

Comment:
ditto.

Reply:
Note how this lint indicates a change to be made rather than an oversight on the linter. Those two kinds of situations are important to distinguish between. See my comments above about linting in general.

Reply:
I understand your point. But I do also consider the priorities of the lib.

I agree, if we can afford reducing the complexity of the types and fix this would be nice. But that should never be in charge of increasing the complexity of the implementation complexity itself. As the algorithm is already complex and difficult sometimes to match to the paper. So if to satisfy clippy and some Rust standards we need to deviate from the implementation style a lot, I would not feel it's worth TBH.

That was my concern when I pointed the clippy lint complaints.